### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by Ghiftee**
Hello!

In this milestone, we implemented the following:

- Install CanCanCan.
- Add a role column to the users table using migration.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
-  Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- Add the "Delete" button to the view and make sure that only authorized users can see it.

Please review our code 😄

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<%= link_to "Create a new post", user_post_new_path(@user.id), class: 'see-posts-button' %>
+<%= link_to "Create a new post", new_post_path(@user.id), class: 'see-posts-button' %>
```
Comment (GzimAsani) [inline app/views/users/show.html.erb:1]: - [x] Unfortunately, I cannot create a new user. After I signup and try to log in I get the `You have to confirm your email address before continuing.` But I don't receive any confirmation email. Please let me know if I am missing something

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @Ghiftee  and @AkashaRojee                             
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb (hunk 1 above)
Comment (Ghiftee) [reply app/views/users/show.html.erb:1]: @GzimAsani thank you for the review, please check your terminal for the link to confirm your email. Once you signup, check the terminal where the rails server runs, you would find the link to confirm your email

Comment (Ghiftee) [reply app/views/users/show.html.erb:1]: There are no issues with the email confirmation that you requested changes for, everything works fine

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**